File: yukicoder/tyama_yukicoder58.py
#!/usr/bin/python
N=input()
K=input()
s=0.
t=7**7
r=__import__('random').randrange
for _ in[0]*t:
	a=0
	for i in range(N):a+=r(1,7)-r(4 if i<K else 1,7)
	s+=a<0
print s/t

File: yukicoder/tyama_yukicoder58.rb
#!/usr/bin/ruby
N,K=$<.map &:to_i
s=0.0
(t=7**7).times{a=0
N.times{|i|a+=rand(1..6);a-=rand((i<K ? 4:1)..6)}
s+=1 if a<0}
p s/t